app/assets/javascripts/controllers/container_deployment/deploy_provider/deploy-provider-config-auth-settings-controller.js

```diff
@@ -0,0 +1,196 @@
+miqHttpInject(angular.module('miq.containers.providersModule')).controller('containers.deployProviderAuthSettingsController',
+  ['$rootScope', '$scope', '$timeout', '$document',
+  function($rootScope, $scope, $timeout, $document) {
+    'use strict';
+    $scope.deploymentDetailsAuthSettingsComplete = false;
+    $scope.reviewTemplate = "/static/deploy-provider-auth-settings-review.html.haml";
+    var firstShow = true;
+    $scope.onShow = function () {
+      if (firstShow) {
+        $scope.data.authentication.htPassword = {
+          users: [
+            {username: '', password: ''}
+          ]
+        };
+        $scope.data.authentication.ldap = {};
+        $scope.data.authentication.requestHeader = {};
+        $scope.data.authentication.openId = {};
+        $scope.data.authentication.google = {};
+        $scope.data.authentication.github = {};
+        firstShow = false;
+      }
+      $scope.validateForm();
+      $timeout(function() {
+        var queryResult = $document[0].getElementById('initial-setting-input');
+        if (queryResult) {
+          queryResult.focus();
+        }
+      }, 200);
+
+      switch ($scope.data.authentication.mode) {
+        case 'htPassword':
+          $scope.authTypeString = 'HTPassword';
+          break;
+        case 'ldap':
+          $scope.authTypeString = 'LDAP';
+          break;
+        case 'requestHeader':
+          $scope.authTypeString = 'Request Header';
+          break;
+        case 'openId':
+          $scope.authTypeString = 'OpenID Connect';
+          break;
+        case 'google':
+          $scope.authTypeString = 'Google';
+          break;
+        case 'github':
+          $scope.authTypeString = 'GitHub';
+          break;
+      }
+    };
+
+    $scope.$watch('data.authentication.mode', function() {
+      $scope.validateForm();
+    });
+
+    var validString = function(value) {
+      return angular.isDefined(value) && value.length > 0;
+    };
+
+    var validHtPassword = function() {
+      if ($scope.data.authentication.mode !== 'htPassword') {
+        return true;
+      }
+
+      if (angular.isUndefined($scope.data.authentication.htPassword)) {
+        return false;
+      }
+
+      var invalid = $scope.data.authentication.htPassword.users.find(function (user) {
+        return !validString(user.username) || !validString(user.password);
+      });
+
+      return invalid === undefined;
+    };
```

Comment:
@zeari do you think the `is` prefix makes it clearer? 
I'm not convinced, but if so I'll need to change other boolean vars in my controllers.